purchases/src/main/kotlin/com/revenuecat/purchases/common/networking/HTTPResult.kt

```diff
@@ -51,6 +52,15 @@ internal data class HTTPResult(
 
     val body: JSONObject = payload.takeIf { it.isNotBlank() }?.let { JSONObject(it) } ?: JSONObject()
 
+    val backendErrorCode: Int? = if (!isSuccessful()) body.optInt("code").takeIf { it > 0 } else null
```

Comment:
What would a zero or negative code mean, or is this just theoretically impossible? Apparent AWS' load balancer [can return 000](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-troubleshooting.html#response-code-000) before it sends a GOAWAY and closes the connection 🤨.

Reply:
These are usually the codes sent by our own servers, which start on `7XXX`. I believe those kind of errors shouldn't return a body with a `code` field like this, so we should be safe from what I can tell. Thanks for checking!